pkg/k8shandler/certificates.go

```diff
@@ -33,30 +33,43 @@ func extractSecretToFile(namespace string, secretName string, key string, toFile
 
 	// check to see if the map value exists
 	if !ok {
-		logrus.Fatalf("No secret data \"%s\" found", key)
-		return nil
+		return fmt.Errorf("No secret data \"%s\" found", key)
 	}
 
 	err = ioutil.WriteFile(path.Join(utils.WORKING_DIR, toFile), value, 0644)
 	if err != nil {
-		logrus.Fatalf("Unable to write to working dir: %v", err)
+		return fmt.Errorf("Unable to write to working dir: %v", err)
 	}
 
 	return nil
 }
 
 func extractMasterCertificate(namespace string, secretName string) error {
 
-	extractSecretToFile(namespace, secretName, "masterca", "ca.crt")
-	extractSecretToFile(namespace, secretName, "masterkey", "ca.key")
+	err := extractSecretToFile(namespace, secretName, "masterca", "ca.crt")
```

Comment:
I think the typical go idiom is to do the assignment in the if statement:
```
if err := extractSecretToFile(namespace, secretName, "masterca", "ca.crt"); err != nil {
      return err
}
```
I can't remember if you can use `:=` here, or have to declare `err` above and use `=` here.

Reply:
:+1: I'm going to play around with using named returns and using that idiom